crates/rpc-types-eth/src/erc4337.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{Log, TransactionReceipt};
+use alloy_primitives::{Address, BlockNumber, Bytes, B256, U256};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// Options for conditional raw transaction submissions.
+// reference for the implementation <https://notes.ethereum.org/@yoav/SkaX2lS9j#>
+// See also <https://pkg.go.dev/github.com/aK0nshin/go-ethereum/arbitrum_types#ConditionalOptions>
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct ConditionalOptions {
+    /// A map of account addresses to their expected storage states.
+    /// Each account can have a specified storage root or explicit slot-value pairs.
+    #[serde(default)]
+    pub known_accounts: HashMap<Address, AccountStorage>,
+    /// The minimal block number at which the transaction can be included.
+    /// `None` indicates no minimum block number constraint.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub block_number_min: Option<BlockNumber>,
+    /// The maximal block number at which the transaction can be included.
+    /// `None` indicates no maximum block number constraint.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub block_number_max: Option<BlockNumber>,
+    /// The minimal timestamp at which the transaction can be included.
+    /// `None` indicates no minimum timestamp constraint.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub timestamp_min: Option<u64>,
+    /// The maximal timestamp at which the transaction can be included.
+    /// `None` indicates no maximum timestamp constraint.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub timestamp_max: Option<u64>,
+}
+
+/// Represents the expected state of an account for a transaction to be conditionally accepted.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum AccountStorage {
+    /// Expected storage root hash of the account.
+    RootHash(B256),
+    /// Explicit storage slots and their expected values.
+    Slots(HashMap<U256, B256>),
+}
+
+/// ERC-4337: User Operation
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UserOperation {
```

Comment:
It would be great to get PackedUserOperation (Entrypoint 0.7) support here. 

The UserOperation struct is (Entrypoint 0.6) is outdated

Reply:
Alright, will update the OP struct with the recent one as mentioned in https://www.erc4337.io/docs

I was consulting the eip documentation so far and it mentioned both structs. Sending the unpacked one returns 32602 err code which means invalid struct, so the packed one makes more sense.